crypto/stark-verifier-ethereum/contracts/stark_verifier.sol

```diff
@@ -0,0 +1,121 @@
+pragma solidity ^0.6.4;
+pragma experimental ABIEncoderV2;
+
+import './interfaces/ConstraintInterface.sol';
+import './public_coin.sol';
+import './proof_of_work.sol';
+import './fri.sol';
+
+
+contract StarkVerifier is ProofOfWork, Fri {
+    using PublicCoin for PublicCoin.Coin;
+
+    // This struct contains all of the componets of the STARK proof.
+    // Please note that any input which would be a 'FieldElement' in rust,
+    // should be the montgomery bytes encoded field element
+    struct StarkProof {
+        // An array with the public inputs to the STARK
+        bytes public_inputs;
+        // An array with the flatened trace table decommitments
+        // For a trace table with n coloums it will be length num_queries*n
+        // and it will be laid out as:
+        // [[query 1 col 1][query 1 col 2]...[query 1 col n]]...[[query q col 1]...[query q col n]]
+        bytes32[] trace_values;
+        // The commitment to those trace values
+        bytes32 trace_commitment;
+        // The trace table evaluated constraint values at the the query indices.
+        bytes32[] constraint_values;
+        // The commitmnet to thos constraint values
+        bytes32 constraint_commitment;
+        // The trace values used for the oods point constraint evaluation
+        bytes32[] trace_oods_values;
+        // The constraint values used for the oods point constraint evaluation
+        bytes32[] constraint_oods_values;
+        // The nonce used for the proof of work
+        bytes8 pow_nonce;
+        // The merkle decomitment for the trace values
+        bytes32[] trace_decommitment;
+        // The merkle decomitment for the constraint evaluated queries
+        bytes32[] constraint_decommitment;
+        // The values to complete each coset of fri at each layer.
+        bytes32[][] fri_values;
+        // The roots for each fri decommitment
+        bytes32[] fri_commitments;
+        // The merkle proof decommitment at each fri layer
+        bytes32[][] fri_decommitments;
+        // The coeffiencts of the last fri layer
+        bytes32[] last_layer_coeffiencts;
+    }
+
+    // This struct contains the relevent information about the constraint system
+    // It will be returned from a callout to the constraint system contract.
```

Comment:
I wouldn't call this 'relevant information about the constraint system'. Maybe call it configuration for the verifier?